### DISCUSSION TITLE
Integrating WatsonTcp with Serilog or Windows Event Viewer

### DISCUSSION THREAD

**Discussion by YGLim1961** [jchristn/WatsonTcp#273] [Q&A] [answered]
I am just starting using WatsonTcp in my project. On the server side, I am starting as a Background Service (with Microsoft Extension Hosting and Configuration).  For logging purposes, is there any step to integrate with
1. Serilog or
2. Windows Event Viewer

Many thanks

**Comment by YGLim1961** [comment]
I found it. Basically just handle this through

   private static void ExceptionEncountered(object sender, ExceptionEventArgs e)
   {
       Console.WriteLine(_Server.SerializationHelper.SerializeJson(e.Exception, true));
   }

Sorry about this.

**Reply by jchristn** [reply]
Glad you got it sorted!

**Reply by YGLim1961** [reply, marked as answer]
Thanks so much. Trying to understand and use this component in an upcoming project.
Regards
